From b252c7aed3fa6f22db8a26c3ab0bfe66e3490eef Mon Sep 17 00:00:00 2001
From: Adrian Perez de Castro <aperez@igalia.com>
Date: Sun, 23 Sep 2018 02:34:26 +0300
Subject: [PATCH xserver] [ARM] Building FELightingNEON.cpp fails due to
 missing lightVector member https://bugs.webkit.org/show_bug.cgi?id=189890

Reviewed by NOBODY (OOPS!).

No new tests needed.

* platform/graphics/cpu/arm/filters/FELightingNEON.h:
(WebCore::FELighting::platformApplyNeon): Adapt to new layout of "struct PaintingData" after r225122.
---
 Source/WebCore/ChangeLog                             | 12 ++++++++++++
 .../graphics/cpu/arm/filters/FELightingNEON.h        |  6 +++---
 2 files changed, 15 insertions(+), 3 deletions(-)

Fetch from: https://bugs.webkit.org/show_bug.cgi?id=189890
Upstream-Status: Pending
Signed-off-by: Adrian Perez de Castro <aperez@igalia.com>

diff --git a/Source/WebCore/platform/graphics/cpu/arm/filters/FELightingNEON.h b/Source/WebCore/platform/graphics/cpu/arm/filters/FELightingNEON.h
index 42af922374b..b542a4c81aa 100644
--- a/Source/WebCore/platform/graphics/cpu/arm/filters/FELightingNEON.h
+++ b/Source/WebCore/platform/graphics/cpu/arm/filters/FELightingNEON.h
@@ -144,9 +144,9 @@ inline void FELighting::platformApplyNeon(const LightingData& data, const LightS
             neonData.flags |= FLAG_CONE_EXPONENT_IS_1;
     } else {
         ASSERT(m_lightSource->type() == LS_DISTANT);
-        floatArguments.lightX = paintingData.lightVector.x();
-        floatArguments.lightY = paintingData.lightVector.y();
-        floatArguments.lightZ = paintingData.lightVector.z();
+        floatArguments.lightX = paintingData.initialLightingData.lightVector.x();
+        floatArguments.lightY = paintingData.initialLightingData.lightVector.y();
+        floatArguments.lightZ = paintingData.initialLightingData.lightVector.z();
         floatArguments.padding2 = 1;
     }
 
-- 
2.19.0

